### PR TITLE
[#3864] feat(lakehouse-iceberg): add custom catalog name to  Iceberg catalog backend

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?
add custom catalog name to Iceberg catalog backend

### Why are the changes needed?

Fix: #3864 #3865 

### Does this PR introduce _any_ user-facing change?
yes, add configuration document

### How was this patch tested?
1.  create a jdbc catalog with custom iceberg-catalog-name
2.  check it takes effects in MySQL 
3.  spark connector could successfully operate under the jdbc catalog.

### CODE REVIEW

**Comment by jerryshao** [comment]
Why do you need to backport to branch-0.5?

**Comment by FANNG1** [comment]
> Why do you need to backport to branch-0.5?

because there is a bug in #3865,  it's also about the iceberg catalog name,  spark connector could operate on Gravitino Iceberg catalog with JDBC backend.

**Comment by jerryshao** [comment]
I don't think there will be a 0.5.2 release anymore.

**Comment by FANNG1** [comment]
> I don't think there will be a 0.5.2 release anymore.

There are known critical bugs in 0.5.1, why not release a bug-fix version?

**Comment by jerryshao** [comment]
Sure, you can create a release if you really want.

**Comment by FANNG1** [comment]
> Sure, you can create a release if you really want.

ok, I think it's necessary and I'd like to do the release

**Comment by FANNG1** [comment]
@jerryshao please help to review this pr again when you have time

File: catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergCatalogOperations.java [hunk 1]
```diff
@@ -90,7 +90,9 @@ public void initialize(
     Map<String, String> resultConf = Maps.newHashMap(prefixMap);
     resultConf.putAll(gravitinoConfig);
     resultConf.put("catalog_uuid", info.id().toString());
-
+    if (resultConf.containsKey(IcebergCatalogPropertiesMetadata.ICEBERG_CATALOG_NAME) == false) {
```
[the remark this answers is not in the document]
Comment (FANNG1) [reply catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergCatalogOperations.java]: done

File: catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergCatalogPropertiesMetadata.java [hunk 2]
```diff
@@ -31,6 +31,7 @@ public class IcebergCatalogPropertiesMetadata extends BaseCatalogPropertiesMetad
   public static final String GRAVITINO_JDBC_DRIVER = "jdbc-driver";
   public static final String WAREHOUSE = "warehouse";
   public static final String URI = "uri";
+  public static final String ICEBERG_CATALOG_NAME = "iceberg-catalog-name";
```
[the remark this answers is not in the document]
Comment (FANNG1) [reply catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergCatalogPropertiesMetadata.java]: It's used to configuration the name of Iceberg catalog backend

File: catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergConfig.java [hunk 3]
```diff
@@ -103,10 +104,21 @@ public class IcebergConfig extends Config implements OverwriteDefaultConfig {
...
+          .doc("The catalog name for Iceberg catalog backend")
+          .version(ConfigConstants.VERSION_0_5_2)
+          .stringConf()
+          .create();
+
   public String getJdbcDriver() {
     return get(JDBC_DRIVER);
   }
 
+  public String getIcebergCatalogName(String defaultCatalogName) {
```
[the remark this answers is not in the document]
Comment (FANNG1) [reply catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergConfig.java]: renamed to `getIcebergCatalogBackendName `

**Review by jerryshao — APPROVED** [review]

**Pull request merged**